packages/strapi-plugin-users-permissions/middlewares/provider/index.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+/**
+ * Module dependencies
+ */
+
+// Public node modules.
+const _ = require('lodash');
+const Grant = require('grant-koa');
+const mount = require('koa-mount');
+
+module.exports = strapi => {
+  return {
+    beforeInitialize: function() {
+      strapi.config.middleware.load.after.push('provider');
+    },
+
+    initialize: function(cb) {
+      _.defaultsDeep(strapi.plugins['users-permissions'].config.grant, {
+        server: {
+          protocol: 'http',
+          host: 'localhost:1337'
+        }
+      });
+
+      const grant = new Grant(strapi.plugins['users-permissions'].config.grant);
+
+      strapi.app.use(mount(grant));
```

Comment:
Did you try without mounting? I'm almost sure we don't need to mount the middleware.

If it doesn't work, try this:

```js
strapi.app.use(strapi.koaMiddlewares.compose(grant.middleware));
```